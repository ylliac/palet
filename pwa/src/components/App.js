import React from 'react'
import logo from '../assets/logo.svg'
import './App.css'
import Image from './Image'
import SelectImageButton from './SelectImageButton'
import LoadSampleButton from './LoadSampleButton'

const App = () => {
  return (
    <div className='App'>
      <div className='App-header'>
        <img src={logo} className='App-logo' alt='logo' />
        <h2>Welcome to React</h2>
      </div>
      <p className='App-intro'>
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      <Image />
      <SelectImageButton />
      <LoadSampleButton />
    </div>
  )
}

export default App
