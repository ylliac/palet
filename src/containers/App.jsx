import React, { Component, PropTypes } from 'react';
import { Provider } from 'react-redux';
import configureStore from '../store/configureStore';

// TODO Should be taken off in production
import DevTools from '../containers/DevTools.jsx';

const store = configureStore();

export default class App extends Component {
  static propTypes = {
    children: PropTypes.node
  };
  render() {
    return (
      <div>
        <Provider store={store}>
          { this.props.children }
        </Provider>
        { __DEVTOOLS__ && <DevTools /> }
      </div>
    );
  }
}
