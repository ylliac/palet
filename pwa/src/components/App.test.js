import React from 'react'
import App from './App'
import renderer from 'react-test-renderer'
import configureMockStore from 'redux-mock-store'
import {Provider} from 'react-redux'

const mockStoreFactory = configureMockStore()

it('renders without crashing', () => {
  const tree = renderer.create(
    <Provider store={mockStoreFactory()}>
      <App />
    </Provider>
    ).toJSON()
  expect(tree).toMatchSnapshot()
})
