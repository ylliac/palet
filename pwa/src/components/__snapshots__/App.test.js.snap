// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.png"
    />
    <h2>
      Palet
    </h2>
  </div>
  <p
    className="App-intro"
  >
    Cliquez sur 'Prendre une photo' ou sur 'Exemple'.
  </p>
  <div />
  <div>
    <label
      htmlFor="selectImageButton"
      style={
        Object {
          "MozAppearance": "push-button",
          "WebkitAppearance": "push-button",
          "appearance": "push-button",
          "lineHeight": "16px",
          "margin": ".2em",
          "padding": ".2em .4em",
        }
      }
    >
      Prendre une photo
    </label>
    <input
      accept="image/*"
      capture="camera"
      id="selectImageButton"
      name="image"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
  </div>
  <button
    onClick={[Function]}
  >
    Exemple
  </button>
</div>
`;
